src/tools/perf/perftest.c

```diff
@@ -296,24 +296,26 @@ static void sock_rte_recv(void *rte_group, unsigned src, void *buffer,
     exit(EXIT_FAILURE);
 }
 
-static void sock_rte_report(void *rte_group, const ucx_perf_result_t *result,
-                            void *arg, const char *extra_info, int is_final,
-                            int is_multi_thread)
+static void rte_report_common(void *rte_group, const ucx_perf_result_t *result,
+                              void *arg, const char *extra_info, int is_final,
+                              int is_multi_thread)
 {
     struct perftest_context *ctx = arg;
     print_progress(ctx->test_names, ctx->num_batch_files, result, extra_info,
                    ctx->flags, is_final, ctx->server_addr == NULL,
                    is_multi_thread);
 }
 
+perftest_rte_report_func_t rte_report_common_func = rte_report_common;
```

Comment:
seems all RTEs use rte_report_common function anyway.. maybe remove .report field altogether, and always use rte_report_common directly for reporting performance (and rename it accordingly)?

Reply:
fixed, still libperf uses passed callback from main perftest as report manipulates `struct perftest_context`.